lib/stack_master/stack_definition.rb

```diff
@@ -41,6 +41,36 @@ def template_dir
     end
 
     def template_file_path
+
+      # Download S3 template if you have skipped upload
+      # This is used for diff/validate commands
+      if s3_configured? && s3['use_remote']
+
+        StackMaster.stdout.puts "I see use_remote flag. Templates will be referenced from S3 bucket and not locally"
+        s3_obj = ::Aws::S3::Client.new(region: s3['region'])
+        raise "Unable to Get S3 driver for downloading template" if s3_obj.nil?
+
+        # Construct file temporary file location for downloading
+        # template from S3 bucket
+        _template_dir = Dir.tmpdir()
+        _template_file_path = File.join(_template_dir, template)
+
+        # Actually download template from S3
+        response = s3_obj.get_object(
+          {
+            bucket: s3['bucket'],
+            key: "#{s3['prefix']}/#{s3_template_file_name}",
+          },
+          target: _template_file_path,
+        ) rescue nil
+
+        if response.nil?
+          StackMaster.stderr.puts "Please double check S3 parameters in your configuration file."
+          raise "Unable to Download #{template} from S3 bucket: #{s3['bucket']}."
+        end
+
+        return _template_file_path
+      end
```

Comment:
Is there a better place for this logic? I wouldn't expect `template_file_path` to have to call S3. Maybe extract methods for the expected remote bucket/key and verify the file is there somewhere else (called from apply).